internal/build/docker.go

```diff
@@ -78,29 +76,27 @@ func (l *localDockerBuilder) BuildDockerFromExisting(ctx context.Context, existi
 
 func (l *localDockerBuilder) buildDocker(ctx context.Context, df Dockerfile,
 	paths []pathMapping, steps []model.Cmd, entrypoint model.Cmd) (digest.Digest, error) {
-	baseDigest, err := l.buildFromDfWithFiles(ctx, df, paths)
+	// Do a docker build with the new files.
+	resultDigest, err := l.buildFromDfWithFiles(ctx, df, paths)
 	if err != nil {
 		return "", fmt.Errorf("buildFromDfWithFiles: %v", err)
 	}
 
-	// Start a container from the base image so we can run steps on it.
-	cID, err := l.pool.claimContainer(ctx, baseDigest)
+	if len(steps) > 0 || !entrypoint.Empty() {
+		// Do a separate docker build with the new steps.
+		stepsDf := DockerfileFromExisting(resultDigest)
+		for _, step := range steps {
+			stepsDf = stepsDf.Run(step)
+		}
 
-	logger.Get(ctx).Verbose("-- executing steps on base image")
-	for i, s := range steps {
-		logger.Get(ctx).Verbose("--- step %d/%d: %v", i+1, len(steps), s.Argv)
-		err := l.execInContainer(ctx, cID, s)
+		stepsDf = stepsDf.Entrypoint(entrypoint)
```

Comment:
I think this will kick our asses if entrypoint is not nil, no? (if we're building from scratch, empty entrypoint should throw an error -- and i believe this will put some nonsense in the dockerfile, or else make the image un-runnable? and if we're building from an existing image, we expect to pass the `buildDocker` helper func an empty entrypoint and use the one from the existing image -- does this break that behavior?)